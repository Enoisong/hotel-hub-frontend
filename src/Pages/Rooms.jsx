import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRooms } from '../redux/features/rooms/roomsSlice';

const Rooms = () => {
  const dispatch = useDispatch();
  const rooms = useSelector((state) => state.room.rooms);
  const isLoading = useSelector((state) => state.room.isLoading);
  const error = useSelector((state) => state.room.error);
  const history = useHistory();

  useEffect(() => {
    dispatch(fetchRooms());
  }, [dispatch]);

  const handleRoomClick = (roomId) => {
    history.push(`/room/${roomId}`);
  };

  if (isLoading) {
    return (<div>Loading...</div>);
  }

  if (error) {
    return (
      <div>
        Error:
        {error}
      </div>
    );
  }

  return (
    <div>
      {rooms.map((room) => (
        <div key={room.id}>
          <p>{room.name}</p>
          <button type="button" onClick={() => handleRoomClick(room.id)}>View Room Details</button>
        </div>
      ))}
    </div>
  );
};

export default Rooms;
