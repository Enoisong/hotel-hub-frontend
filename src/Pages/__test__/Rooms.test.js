// Import the necessary dependencies
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom'; // Add this line
import Rooms from '../Rooms';

// Mock the redux store
const mockStore = configureStore([]);

describe('Rooms Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      room: {
        rooms: [],
        error: null,
      },
    });
  });

  it('renders without errors', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          {' '}
          {/* Wrap with BrowserRouter */}
          <Rooms />
        </BrowserRouter>
      </Provider>,
    );

    // Your test assertions
  });

  // ...

  it('renders without errors', () => {
    // Mock useNavigate
    const navigate = jest.fn();

    // Render the component with the mock navigate function
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Rooms navigate={navigate} />
        </BrowserRouter>
      </Provider>,
    );

    // Your test assertions
  });

  // Add other test cases as needed

  // Add a snapshot test
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rooms />
        </BrowserRouter>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
