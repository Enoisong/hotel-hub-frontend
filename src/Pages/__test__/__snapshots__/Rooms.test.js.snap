// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Please log in to view available rooms.
    <button
      type="button"
    >
      Click here
    </button>
  </div>
</DocumentFragment>
`;
